src/compile/mark/valueref.ts

```diff
@@ -83,18 +75,22 @@ export function midPoint(channel: Channel, channelDef: ChannelDef, scaleName: st
   if (channelDef) {
     /* istanbul ignore else */
     if (isFieldDef(channelDef)) {
+      if (scale.type === 'bin-linear') {
+        return {
+          signal: `(scale("${scaleName}", ${field(channelDef, {binSuffix: 'start', datum: true})}) + scale("${scaleName}", ${field(channelDef, {binSuffix: 'end', datum: true})}))/2`
```

Comment:
The problem is that it only works for certain channels and making it a separate function hides the implementation.